src/libstd/fs.rs

```diff
@@ -1536,6 +1531,28 @@ mod tests {
         TempDir(ret)
     }
 
+    // Several test fail on windows if the user does not have permission to create symlinks (the
+    // `SeCreateSymbolicLinkPrivilege`). Instead of disabling these test on Windows, use this
+    // function to test whether we have permission, and return otherwise. This way, we still don't
+    // run these tests most of the time, but at least we do if the user has the right permissions.
+    #[cfg(windows)]
+    pub fn got_symlink_permission(tmpdir: &TempDir) -> bool {
+        let link = tmpdir.join("some_hopefully_unique_link_name");
+
+        match symlink_file(r"nonexisting_target", link) {
+            Ok(_) => true,
+            Err(ref err) =>
+                if err.to_string().contains("A required privilege is not held by the client.") {
+                    false
+                } else {
+                    true
+                }
+        }
+    }
+    #[cfg(not(windows))]
+    #[allow(unused_variables)]
```

Comment:
Instead of conditional compilation here (e.g. defining the same signature twice and having an `#[allow]` this could also do something like

```rust
if cfg!(unix) {
    return true
}
```

That should be more resilient to cross-compilation concerns in the future at least